applications/annotate_points.md

```diff
@@ -0,0 +1,383 @@
+        # add the label menu widget to the viewer
+        viewer.window.add_dock_widget(label_menu)
+
+        @viewer.bind_key('.')
+        def next_label(event=None):
+            """Keybinding to advance to the next label with wraparound"""
+            current_properties = points_layer.current_properties
+            current_label = current_properties['label'][0]
+            ind = list(labels).index(current_label)
+            new_ind = (ind + 1) % len(labels)
+            new_label = labels[new_ind]
+            current_properties['label'] = np.array([new_label])
+            points_layer.current_properties = current_properties
+
+        def next_on_click(layer, event):
+            """Mouse click binding to advance the label when a point is added"""
+            if layer.mode == 'add':
+                next_label()
+
+                # by default, napari selects the point that was just added
+                # disable that behavior, as the highlight gets in the way
+                layer.selected_data = {}
+
+        points_layer.mouse_drag_callbacks.append(next_on_click)
+
+
+        @viewer.bind_key(',')
+        def prev_label(event):
+            """Keybinding to decrement to the previous label with wraparound"""
+            current_properties = points_layer.current_properties
+            current_label = current_properties['label'][0]
+            ind = list(labels).index(current_label)
+            n_labels = len(labels)
+            new_ind = ((ind - 1) + n_labels) % n_labels
+            new_label = labels[new_ind]
+            current_properties['label'] = np.array([new_label])
+            points_layer.current_properties = current_properties
+
+
+        @viewer.bind_key('Control-S')
+        def save_points(event):
+            """Save the added points to a CSV file"""
+            # get the frame indices
+            frame_indices = np.unique(points_layer.data[:, 0]).astype(np.int)
+
+            # get the filenames
+            all_files = np.asarray(glob.glob(im_path))
+            file_names = all_files[frame_indices]
+
+            # create and write dataframe
+            header = pd.MultiIndex.from_product(
+                [[scorer], labels, ['x', 'y']],
+                names=['scorer', 'bodyparts', 'coords']
+            )
+            df = pd.DataFrame(
+                index=file_names,
+                columns=header,
+            )
+
+            # populate the dataframe
+            for label, coord in zip(points_layer.properties['label'], points_layer.data):
+                fname = all_files[coord[0].astype(np.int)]
+                df.loc[fname][scorer][label]['x'] = coord[2]
+                df.loc[fname][scorer][label]['y'] = coord[1]
+
+            # write the dataframe
+            df.to_csv(output_path)
+        
+```
+## point_annotator()
+We will create the GUI within a function called `point_annotator()`. Wrapping the GUI creation in the function allows us to  See below for the function definition.
```

Comment:
Something has gone wrong with this sentence, end cut off maybe

Reply:
Good catch. Fixed!